src/v/cloud_storage/partition_manifest.cc

```diff
@@ -1008,6 +1028,8 @@ struct partition_manifest_handler
     std::optional<model::offset_delta> _delta_offset_end;
     std::optional<segment_name_format> _meta_sname_format;
 
+    ss::shared_ptr<util::mem_tracker> _manifest_mem_tracker;
```

Comment:
I haven't gone into the impl of mem_tracker: are you confident it doesn't have any significant overhead?

Reply:
Fairly confident, under the hood it's just a list of child trackers (of which there are none here), a counter, and a label.

It'll be interesting to see how mem-tracking evolves in the codebase, but this seems like a great place to start considering the ongoing memory-focused efforts.